### ISSUE TITLE
WSL 2 distro connection broken after each update

### ISSUE THREAD

**Issue by Vincent-Noben** [microsoft/vscode#204081]
Type: <b>Bug</b>

Every time VS code releases an update my connection to my WSL 2 is broken. To fix it I have to remove the .vscode-server folder on my WSL machine and start vscode again. But than all extentions are disabled again and I have to setup it all over again. This has happened multiple times now and is a pain in the a.

Note: my WSL setup is not on the default C drive location but has been moved to the D drive in the usual manner. 

VS Code version: Code 1.86.0 (05047486b6df5eb8d44b2ecd70ea3bdf775fd937, 2024-01-31T10:28:19.990Z)
OS version: Windows_NT x64 10.0.19045
Modes:
Connection to 'WSL: Ubuntu-22.04-2' could not be established  VS Code Server for WSL closed unexpectedly.

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|12th Gen Intel(R) Core(TM) i9-12900H (20 x 2918)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|31.68GB (15.27GB free)|
|Process Argv|-n --crash-reporter-id 657ee562-e9b9-4d4c-b83c-467d809d92a0|
|Screen Reader|no|
|VM|0%|

Connection to 'WSL: Ubuntu-22.04-2' could not be established  VS Code Server for WSL closed unexpectedly.
</details>Extensions: none<details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vspor879:30202332
vspor708:30202333
vspor363:30204092
vscorecescf:30445987
vscod805cf:30301675
binariesv615:30325510
vsaa593cf:30376535
py29gd2263:30899288
c4g48928:30535728
azure-dev_surveyone:30548225
2i9eh265:30646982
0bi6i642:30951795
pythongtdpath:30769146
welcomedialogc:30910334
pythonidxpt:30866567
pythonnoceb:30805159
asynctok:30898717
pythontestfixt:30902429
pythonregdiag2:30936856
pyreplss1:30897532
pythonmypyd1:30879173
pythoncet0:30885854
pythontbext0:30879054
accentitlementsc:30887149
dsvsc016:30899300
dsvsc017:30899301
dsvsc018:30899302
dsvsc019b:30953937
3ef8e399:30949928

```

</details>

**Comment by aeschli** [comment]
When you say the connection is broken, do you mean the opening the WSL window does not work. Can you share the WSL log? (Command: `WSL: Show Log...`)
If the fix is to delete the `.vscode-server`, that could mean the server installs are corrupted. Note: the server installs are in `.vscode-server/bin`, no need to delete all of `.vscode-server`,

What do you mean by WSL setup on the D drive?

**Comment by Vincent-Noben** [comment]
Hi @aeschli, thanks for your reply.

My WSL log:
[2024-02-02-wsl-log.log](https://github.com/microsoft/vscode/files/14137459/2024-02-02-wsl-log.log)
(Was after doing my "remove .vscode-server folder" fix and after restarting vscode, so not sure if it contains the issue)

> When you say the connection is broken, do you mean the opening the WSL window does not work.

What I mean is that when I open a project in my WSL2 remote I get a popup saying something like "Could not connect to remote" with buttons "ignore, reopen window, ignore" or something similar to that. (Can't reproduce now as I already did my "remove .vscode folder" fix)

Thanks for the tip regarding `.vscode-server/bin, no need to delete all of .vscode-server`, will note that and try if the fault happens again.

> What do you mean by WSL setup on the D drive?

I mean that my C drive is rather small and that I moved my WSL distro to my larger D drive using the documentation found in the docs about WSL. Was a long time ago so not exactly sure how it was done but I assume using wsl export/import commands as shown here: https://learn.microsoft.com/en-us/windows/wsl/faq#how-can-i-move-my-wsl-distribution-to-a-different-drive-or-location-
